### PR TITLE
Correct some accidental renames

### PR DESCRIPTION
**Pull request by bjchambers**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [*] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [*] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [*] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [*] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

IDE over-eagerly replaced some occurrences of createAggregator with
createAggregatorForDoFn. This corrects that.

### CODE REVIEW

**Comment by bjchambers** [comment]
R: @tgroh @dhalperi

**Comment by tgroh** [comment]
LGTM.

File: sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Aggregator.java [hunk 1]
```diff
@@ -81,15 +81,15 @@
      *  <p>This method is called to create an aggregator for a {@link DoFn}. It receives the class
      *  of the {@link DoFn} being executed and the context of the step it is being executed in.
      */
-    <InputT, AccumT, OutputT> Aggregator<InputT, OutputT> createAggregatorForDoFn(
+    <InputT, AccumT, OutputT> Aggregator<InputT, OutputT> createAggregator(
```
Comment (dhalperi) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/transforms/Aggregator.java]: this one needs revert I believe.

**Comment by dhalperi** [comment]
Also now needs a merge on top of Kenn's magic.

**Comment by bjchambers** [comment]
Rebased.

**Comment by dhalperi** [comment]
Build failing:

```
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /home/jenkins/jenkins-slave/workspace/beam_PreCommit_MavenVerify/sdks/java/core/src/main/java/org/apache/beam/sdk/util/CounterAggregator.java:[43,18] org.apache.beam.sdk.util.CounterAggregator.CounterAggregatorFactory is not abstract and does not override abstract method <InputT,AccumT,OutputT>createAggregator(java.lang.Class<?>,org.apache.beam.sdk.util.ExecutionContext.StepContext,java.lang.String,org.apache.beam.sdk.transforms.Combine.CombineFn<InputT,AccumT,OutputT>) in org.apache.beam.sdk.transforms.Aggregator.AggregatorFactory
[ERROR] /home/jenkins/jenkins-slave/workspace/beam_PreCommit_MavenVerify/sdks/java/core/src/main/java/org/apache/beam/sdk/util/CounterAggregator.java:[50,5] method does not override or implement a method from a supertype
[INFO] 2 errors 
```

**Comment by dhalperi** [comment]
Build still failing:

```
[ERROR] COMPILATION ERROR : 
[INFO] -------------------------------------------------------------
[ERROR] /home/jenkins/jenkins-slave/workspace/beam_PreCommit_MavenVerify/runners/core-java/src/main/java/org/apache/beam/sdk/util/DoFnRunnerBase.java:[348,31] cannot find symbol
  symbol:   method createAggregatorForDoFn(java.lang.Class<capture#1 of ? extends org.apache.beam.sdk.transforms.OldDoFn>,org.apache.beam.sdk.util.ExecutionContext.StepContext,java.lang.String,org.apache.beam.sdk.transforms.Combine.CombineFn<AggInputT,capture#2 of ?,AggOutputT>)
  location: variable aggregatorFactory of type org.apache.beam.sdk.transforms.Aggregator.AggregatorFactory
[INFO] 1 error
```

**Comment by dhalperi** [comment]
Still pending, looks like?

**Pull request closed without merging**